pyarn/parser.py

```diff
@@ -57,13 +57,25 @@ def p_block_comment(p):
 def p_title(p):
     """title : STRING COLON INDENT
              | list COLON INDENT"""
+    if type(p[1]) == str:
+        if ',' in p[1]:
+            raise ValueError(f'Following key has a quoted comma: "{p[1]}"')
+    elif type(p[1]) == list:
+        for key in p[1]:
+            if ',' in key:
+                raise ValueError(f'Following key has a quoted comma: "{key}"')
+        p[1] = ', '.join(p[1])
     p[0] = p[1]
 
 
 def p_list(p):
     """list : STRING COMMA STRING
             | list COMMA STRING"""
-    p[0] = ', '.join([p[1], p[3]])
+    if type(p[1]) == str:
+        p[0] = [p[1], p[3]]
+    elif type(p[1]) == list:
+        p[1].append(p[3])
+        p[0] = p[1]
```

Comment:
Minor nitpick: the more "pythonic" way to check types is via isinstance, e.g. `isinstance(x, list)`

And since this parser has only two cases, I would prefer `if-else` over `if-elif-<nothing>`. Or `if-elif-else` and raising an error in the else branch, but that seems unnecessary